### PR TITLE
Adds github omniauth

### PR DESCRIPTION
**Pull request by skuhlmann**
Uses omniauth github gem
Adds session controller with create and destroy methods
Adds create method on user model
Adds users controller with new method
Adds routes for users#new page, create, destroy
Adds controller and feature tests

### CODE REVIEW

File: config/initializers/omniauth.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Rails.application.config.middleware.use OmniAuth::Builder do
+  provider :github, ENV['GITHUB_CLIENT_ID'], ENV['GITHUB_CLIENT_SECRET'], scope: "user:email,public_repo"
```
Comment (skuhlmann) [inline config/initializers/omniauth.rb]: This is an initializer. I'm ignoring it.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -29,3 +30,15 @@
...
   Kernel.srand config.seed
 end
+
+OmniAuth.config.test_mode = true
+omniauth_hash = { "provider" => "github",
+                  "uid" => "12345",
+                  "info" => {
+                    "name" => "jimmy",
+                    "email" => "jimmy@example.com",
+                    "image" => "https://avatars2.githubusercontent.com/u/6923345?v=3&s=460"
```
Comment (skuhlmann) [inline spec/spec_helper.rb]: boo hoo hound.

File: config/initializers/omniauth.rb (hunk 1 above)
Comment (trayo) [inline config/initializers/omniauth.rb]: I don't mind for this either.

**Comment by chandracarney** [comment]
This GitHub auth setup looks good

**Pull request merged**